plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/controls/ViewerColumnController.java

```diff
@@ -840,7 +841,17 @@ public void handleEvent(Event e) {
             Item column = (Item)e.widget;
             if (prevColumn == column) {
                 // Set reverse order
-                sortDirection = sortDirection == SWT.UP ? SWT.DOWN : SWT.UP;
+                if (viewer.getComparator() == null) {
+                    viewer.setComparator(comparator);
+                }
```

Comment:
Why are you removing the comparator? Just toggle between UP/DOWN/NONE sort directions.

Reply:
setting to NONE doesn't help